osfoffline/application/background.py

```diff
@@ -1,103 +1,82 @@
+from osfoffline.polling_osf_manager import polling
+
+
+logger = logging.getLogger(__name__)
 
 
 class BackgroundWorker(threading.Thread):
+
     def __init__(self):
         super().__init__()
+
+        # Start out with null variables for NoneType errors rather than Attribute Errors
         self.user = None
-        self.osf_folder = ''
+        self.osf_folder = None
+
         self.loop = None
-        self.paused = True  # start out paused
-        self.running = False
         self.poller = None
         self.observer = None
 
-    def run(self):
-
-        logging.debug('Run in background tasks called for first time.')
-        self.loop = self.ensure_event_loop()
-        self.run_background_tasks()
-        self.loop.run_forever()
-
-    def run_background_tasks(self):
-        logging.debug('starting run_background_tasks')
+    # courtesy of waterbutler
+    def ensure_event_loop(self):
+        """Ensure the existance of an eventloop
+        Useful for contexts where get_event_loop() may raise an exception.
+        Such as multithreaded applications
 
-        if not self.running:
-            self.user = self.get_current_user()
-            self.osf_folder = self.user.osf_local_folder_path
+        :returns: The new event loop
+        :rtype: BaseEventLoop
+        """
+        try:
+            return asyncio.get_event_loop()
+        except (AssertionError, RuntimeError):
+            asyncio.set_event_loop(asyncio.new_event_loop())
 
-            logging.debug('start observing')
-            self.start_observing_osf_folder()
-            logging.debug('start polling')
-            self.start_polling_server()
-            self.running = True
+        # Note: No clever tricks are used here to dry up code
+        # This avoids an infinite loop if settings the event loop ever fails
+        return asyncio.get_event_loop()
 
-    def pause_background_tasks(self):
+    def run(self):
+        logger.debug('Background worker starting')
+        self.loop = self.ensure_event_loop()
+        # self.loop.set_debug(True)
 
-        if self.running:
-            self.stop_polling_server()
-            self.stop_observing_osf_folder()
-            # self.stop_loop()
-            self.running = False
+        self.user = self.get_current_user()
+        self.osf_folder = self.user.osf_local_folder_path
 
-    def start_polling_server(self):
-        self.poller = polling.Poll(self.user, self.loop)
-        self.poller.start()
+        logger.debug('Starting observer thread')
+        self.start_folder_observer()
 
-    def stop_polling_server(self):
-        if self.poller:
-            self.poller.stop()
+        logging.debug('Starting OSF polling')
+        self.start_osf_poller()
 
-    # todo: can refactor this code out to somewhere
+        logging.debug('Starting background event loop')
+        try:
+            self.loop.run_forever()
+        except Exception as e:
+            print(e)
```

Comment:
Use `logging.exception('Background worker failed unexpectedly')` instead- if the polling fails this hard, we should make sure to capture the traceback.

(are there circumstances where an uncaught exception stopping the loop would be normal? The bare `except Exception as e` is a bit vague wrt expected failure case)